demos/main.js

```diff
@@ -4,74 +4,65 @@
+                url: 'http://vector.dev.mapzen.com/osm/all/{z}/{x}/{y}.json'
             },
-            scene: 'demos/styles.yaml'
-        },
-        'mapzen-local': {
-            source: {
+            'mapzen-local': {
                 type: 'GeoJSONTileSource',
                 url: 'http://localhost:8080/all/{z}/{x}/{y}.json'
             },
-            scene: 'demos/styles.yaml'
-        },
-        'mapzen-mvt': {
-            source: {
+            'mapzen-mvt': {
                 type: 'MapboxFormatTileSource',
-                url: appendProtocol('//vector.mapzen.com/osm/all/{z}/{x}/{y}.mapbox')
+                url: 'http://vector.mapzen.com/osm/all/{z}/{x}/{y}.mapbox'
             },
-            scene: 'demos/styles.yaml'
-        },
-        'mapzen-topojson': {
-            source: {
+            'mapzen-topojson': {
                 type: 'TopoJSONTileSource',
-                url: appendProtocol('//vector.mapzen.com/osm/all/{z}/{x}/{y}.topojson')
-            },
-            scene: 'demos/styles.yaml'
-        }//,
-        // 'osm': {
-        //     source: {
-        //         type: 'GeoJSONTileSource',
-        //         url: 'http://tile.openstreetmap.us/vectiles-all/{z}/{x}/{y}.json'
-        //     },
-        //     scene: 'demos/styles.yaml'
-        // },
-        // 'mapbox': {
-        //     source: {
-        //         type: 'MapboxFormatTileSource',
-        //         url: 'http://{s:[a,b,c,d]}.tiles.mapbox.com/v4/mapbox.mapbox-streets-v6-dev/{z}/{x}/{y}.vector.pbf?access_token=pk.eyJ1IjoiYmNhbXBlciIsImEiOiJWUmh3anY0In0.1fgSTNWpQV8-5sBjGbBzGg',
-        //         max_zoom: 15
-        //     },
-        //     scene: 'demos/styles.yaml'
-        // }
-    };
+                url: 'http://vector.mapzen.com/osm/all/{z}/{x}/{y}.topojson'
+            }
+
+    }, default_tile_source = 'mapzen',
+        scene_url = 'demos/styles.yaml',
+        osm_debug = false,        
+        locations = {
+            'London': [51.508, -0.105, 15],
+            'New York': [40.70531887544228, -74.00976419448853, 16],
+            'Seattle': [47.609722, -122.333056, 15]
+        }, rS, url_hash, map_start_location, url_ui, url_style;
+
+
+
+    getVaulesFromUrl();
+
+    // default source, can be overriden by URL
+    var 
+        map = L.map('map', {
+            maxZoom: 20,
+            trackResize: true,
+            inertia: false,
+            keyboard: false
+        }),
+
+        layer = Tangram.leafletLayer({
+            scene: scene_url,
+            preUpdate: preUpdate,
+            postUpdate: postUpdate,
+            logLevel: 'debug',
+            attribution: 'Map data &copy; OpenStreetMap contributors | <a href="https://github.com/tangrams/tangram" target="_blank">Source Code</a>'
+        });
+
+    layer.scene.subscribe({
+        loadScene: function (config) {
+            config.sources['osm'] = tile_sources[default_tile_source];
```

Comment:
If we move styles under sources, I guess we could still make this work by just updating specific keys instead of overwriting the whole object, yes?